main.py

```diff
@@ -0,0 +1,160 @@
+
+def get_username_and_password(auth_header):
+    # Remove the "Basic " prefix.
+    auth_header = request.headers.get('authorization')[6:]
+    app.logger.info('auth header = [%s]', auth_header)
+
+    username, password = base64.b64decode(auth_header).decode('utf-8').split(':')
+    app.logger.info('username = [%s], password = [%s]', username, password)
+
+    return username, password
+
+def make_irods_credentials_dict(username, password):
+    return {
+        'host': 'localhost',
+        'port': 1247,
+        'user': username,
+        'password': password,
+        'zone': 'tempZone'
+    }
+
+def generate_token(data):
+    #key = jwk.JWK(generate='oct', size=256)
+    key = jwk.JWK.from_password(IRODS_TICKET_BOOTH_SHARED_SECRET)
+
+    token = jwt.JWT(header={'alg': 'HS256'}, claims=data)
+    token.make_signed_token(key)
+    return token.serialize()
+
+    # An encrypted token. This may be helpful in the final implementation.
+    #etoken = jwt.JWT(header={'alg': 'A256KW', 'enc': 'A256CBC-HS512'}, claims=token.serialize())
+    #etoken.make_encrypted_token(key)
+    #return etoken.serialize()
+
+@app.route("/create")
+def create():
+    if 'authorization' not in request.headers:
+        return 'Authorization header is not set.'
+
+    username, password = get_username_and_password(request.headers.get('authorization'))
+
+    if 'collection' not in request.args:
+        return '"collection" argument is not set.'
+
+    # Verify that the collection exists.
+    collection = Markup(request.args.get('collection', '')).unescape()
+    app.logger.info('collection = [%s]', collection)
+
+    # Permissions default to 'read'.
+    permission = 'read'
+    if 'permission' in request.args:
+        permission = Markup(request.args.get('permission', '')).unescape()
+        if permission not in ['read', 'write']:
+            return 'Invalid ticket permission [{}]'.format(permission)
+
+    with iRODSSession(**make_irods_credentials_dict(username, password)) as session:
+        if not session.collections.exists(collection):
+            return "Collection does not exist or user does not have permission to access the collection"
+
+        # Generate a ticket with the requested permissions for a collection.
+        # TODO What happens if the ticket matches a preexisting token?
+        ticket_handle = Ticket(session)
+        ticket_handle.issue(permission, collection)
+
+        # Generate a JWT containing the ticket and the collection associated with
+        # the ticket.
+        return generate_token({'ticket': ticket_handle.ticket, 'collection': collection})
+
+@app.route("/resolve/<token>")
+def resolve(token):
+    app.logger.info('token = [%s]', token)
+
+    # Extract the payload from the JWT and return it.
+    key = jwk.JWK.from_password(IRODS_TICKET_BOOTH_SHARED_SECRET)
+    payload = jwt.JWT(key=key, jwt=token)
+    app.logger.info('JWT claims = [%s]', payload.claims)
+
+    return payload.claims
+
```

Comment:
what is the use case for this?   the requestor should already have the shared secret and can do this themselves?

Reply:
This op does not require any authentication. It can be used to see what the JWT contains assuming the JWT was produced by this app. I'm not sure how people will use this, but it could have some value.

Reply:
could be - let's comment it out for now, need to think through a use case or ramifications a bit more.

Reply:
done